### ISSUE TITLE
Shell script function `intToBin` seems to be incompatible with Debian-based systems

### ISSUE THREAD

**Issue by gsnoff** [IceDBorn/pipewire-screenaudio#26]
The following line:
```Bash
    xargs -I {} sh -c "printf '{}'" # Return raw bytes
```
seems semantically ambiguous in terms of how an escaped stdin sequence should be treated inside the `"printf '{}'"` argument. Apparently, on some systems, `sh`’s builtin `printf` command is capable of unescaping the format string on its own, while on others (particularly where `sh` is a symlink to `dash`) it is not.

(For reference, POSIX standard only defines octal `\0NN`/`\1NN`/`\2NN`/`\3NN` sequences for shell’s `printf`, but not hexadecimal `\xNN` ones.)

Please investigate this in more detail, and implement a solution which does not rely on such obscure semantic intricacies (for example, by using `xxd -r -p -` stage).

**Comment by jim3692** [comment]
Thank you for pointing this issue. You are correct about the ambiguity of `printf` on `sh`. This is because `sh` may by a symlink to either `ash`, `bash` or `dash`, depending on the distro. Unfortunately we did not take this into consideration when we developed those functions.

Sadly we cannot use `xxd` at the moment, although it would simplify some parts of the code. The reason is that `xxd` may be bundled with `vim` on some distros, including Arch.

Specifically for NeoVim users on Arch, there is [neovim-symlinks](https://aur.archlinux.org/packages/neovim-symlinks), which conflicts with `vim`. Maybe we need to check whether [neovim-vim-compat](https://aur.archlinux.org/packages/neovim-vim-compat) and/or [xxd-standalone](https://aur.archlinux.org/packages/xxd-standalone) can be used to solve this problem.
